### DISCUSSION TITLE
ANY failed, tried 4 builders

### DISCUSSION THREAD

**Discussion by hwz9612** [strongswan/strongswan#2223] [Q&A]
When I execued the following command,
`pki --self --ca --lifetime 3652 --in strongswanKey.pem --dn "C=CH, O=strongSwan, CN=strongSwan Root CA" --outform pem > strongswanCert.pem`
The error exists:
`building CRED_PRIVATE_KEY - ANY failed, tried 4 builders
loading private key failed`
Why is this happening?

**Comment by tobiasbrunner** [comment]
The key you try to load there is probably invalid. Going by the previous version of your post, this can happen if you don't have a plugin loaded that can generate a particular type of key (e.g. `ed25519` requires either the _curve25519_ plugin or one of the plugins that link third-party crypto libraries, like the _openssl_ plugin). Use a different key type (e.g. `ecdsa` or `rsa`) or load a plugin that supports the one you want.
